cmd/kube-apiserver/app/server.go

```diff
@@ -244,14 +245,22 @@ func CreateNodeDialer(s completedServerRunOptions) (tunneler.Tunneler, *http.Tra
 		// Get ssh key distribution func, if supported
 		var installSSHKey tunneler.InstallSSHKey
 
-		cloudprovider.DeprecationWarningForProvider(s.CloudProvider.CloudProvider)
-		cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
-		if err != nil {
-			return nil, nil, fmt.Errorf("cloud provider could not be initialized: %v", err)
+		if utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) && cloudprovider.IsDeprecatedInternal(s.CloudProvider.CloudProvider) {
+			return nil, nil, fmt.Errorf(
+				"cloud provider %q was specified, but in-tree cloud providers are disabled. Please move to --cloud-provider=external and out-of-tree CCM/CSI in your cluster",
+				s.CloudProvider.CloudProvider)
 		}
-		if cloud != nil {
-			if instances, supported := cloud.Instances(); supported {
-				installSSHKey = instances.AddSSHKeyToAllInstances
+
+		if !cloudprovider.IsExternal(s.CloudProvider.CloudProvider) {
```

Comment:
I think it's safe to remove this if, there was no special handling for external prior to this